lnrpc/routerrpc/forward_interceptor.go

```diff
@@ -193,8 +195,66 @@ func (r *forwardInterceptor) resolveFromClient(
 	switch in.Action {
 	case ResolveHoldForwardAction_RESUME:
 		return interceptedForward.Resume()
+
 	case ResolveHoldForwardAction_FAIL:
-		return interceptedForward.Fail()
+		// Fail with an encrypted reason.
+		if in.FailureMessage != nil {
+			if in.FailureCode != 0 {
+				return status.Errorf(
+					codes.InvalidArgument,
+					"failure message and failure code "+
+						"are mutually exclusive",
+				)
+			}
+
+			// Verify that the size is equal to the fixed failure
+			// message size + hmac + two uint16 lengths. See BOLT
+			// #4.
+			if len(in.FailureMessage) !=
+				lnwire.FailureMessageLength+32+2+2 {
+
+				log.Debugf("Message length: %v", len(in.FailureMessage))
```

Comment:
nit: consider adding a bit more detail to the log message, "message lenght N" is a bit hard to understand when debugging.

Reply:
Ah that was a left over development log line, removed.